### PR TITLE
ArrayList(u8) support writer interface

### PR DESCRIPTION
**Pull request by marler8997**
Deprecates `outStream` for `writer`.  Also puts u8-specific functions inside a `u8` condition.  Also adds a test for the writer interface.

### CODE REVIEW

**Comment by joachimschmidt557** [comment]
Would it make sense to add `Deprecated` to the docstring of the `ArrayList(T).outStream` function then?

File: lib/std/array_list.zig [hunk 1]
```diff
@@ -50,6 +50,19 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
             self.allocator.free(self.allocatedSlice());
         }
 
+        pub usingnamespace if (T != u8) struct { } else struct {
+            const WriteReturn = @typeInfo(@TypeOf(write)).Fn.return_type.?;
+            pub const Error = @typeInfo(WriteReturn).ErrorUnion.error_set;
```
Comment (daurnimator) [inline lib/std/array_list.zig:55]: No need to be fancy, we know its just `OutOfMemory`

**Comment by marler8997** [comment]
@joachimschmidt557  yes, I didn't realize it already supported `outStream`.  I've fixed that.

File: lib/std/array_list.zig (hunk 1 above)
Comment (marler8997) [reply lib/std/array_list.zig]: ok

**Pull request merged**